pkg/network/testutil/server.go

```diff
@@ -111,6 +113,16 @@ func StartServerUDP(t *testing.T, ip net.IP, port int) io.Closer {
 	}
 
 	udpConn, err := net.ListenUDP(network, addr)
+	assert.Nil(t, err)
+
+	if port == 0 {
+		addrStr := udpConn.LocalAddr().String()
+		_, portStr, err := net.SplitHostPort(addrStr)
+		assert.Nil(t, err)
+		port, err = strconv.Atoi(portStr)
+		assert.Nil(t, err)
+	}
```

Comment:
Do we need to check for `port == 0` ?  This codepath would work both ways

Reply:
I guess it does, although it is only needed when port=0... I removed for more clarity though :)